### PR TITLE
Add ROH, UPD and Cytoband Images to 'case.html'

### PR DESCRIPTION
**Pull request by mikaell**
** Replaces PR #1291 **
** Replaces branch "feature_cytobands_190708" **

Add a new image panel to case.html showing coverage, roh and
chromosome ideograms. Configurable with:

```
chromograph_image_files: scout/demo/images
chromograph_prefixes:
    'roh': "roh_chr"
    'upd': "upd_chr"
    'chr': "cytoband.txt.chr"
```

Feautes:
* Support for Bootstrap 4
* Somewaht dynamic resizing of image panel
* Checks for files and config before attempting to add the content
* etc

This PR adds a functionality or fixes a bug.

**How to test**:
1. how to install it
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by dNil
- [x] tests executed by dNil

### CODE REVIEW

**Comment by northwestwitch** [comment]
I'm reviewing this PR, but it's quite hard to figure out which the final effect might be without actually having pictures at all. 
Could you  either add the required pictures to the demo or simply give me some test pictures to test the code with?

**Comment by dnil** [comment]
Hi @northwestwitch! I have some (old?) files to test with if you want - or I could do the test and post a couple of pictures?

**Comment by northwestwitch** [comment]
> Hi @northwestwitch! I have some (old?) files to test with if you want - or I could do the test and post a couple of pictures?

If you have the pics you could do do the test, and the review I guess?

**Comment by dnil** [comment]
Here's a picture on a working demo case! Really fun to see!

Screenshot 2019-10-22 at 12 34 02
A few quick issues - will have to run to a meeting in a few mins. Will return to this later!

- [ ] the edit button doesn't really belong there?
- [ ] the dynamic sidebar doesn't play too well with this panel - it is all too happy to stay on top instead of forcing its space and the panel to shrink/rearrange into fewer columns.
- [ ] there is some issue when opening a case w/o the chromograph path info; the script section croaks, as if it cannot handle a None value for some of that.
- [ ] spacing between the images and the card title is too small / nonexistent.

**Comment by mikaell** [comment]
I'll fix the according to feedback and get back during this week.

File: scout/server/blueprints/cases/views.py [hunk 1]
```diff
@@ -862,3 +862,31 @@ def multiqc(institute_id, case_name):
...
+    return host_image_aux(institute_id, case_name, image, 'chr_images')
+
+
+def host_image_aux(institute_id, case_name, image, imgstr):
+    """Auxilary function for generate absolute file paths"""
+    institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
+    chr_path = case_obj.get('chromograph_image_files')
+    abs_path = os.path.abspath(chr_path)
+    img_path = abs_path +"/" + imgstr
+    LOG.info("Attempting to send {}/{}" .format(img_path, image))
```
[the remark this answers is not in the document]
Comment (mikaell) [reply scout/server/blueprints/cases/views.py]: Fixed!

**Comment by mikaell** [comment]
> * [ ]  there is some issue when opening a case w/o the chromograph path info; the script section croaks, as if it cannot handle a None value for some of that.

I can't reproduce this. What settings/lack of settings did you use?

File: scout/server/blueprints/cases/templates/cases/case.html [hunk 2]
```diff
@@ -288,6 +293,21 @@ <h4 class="mt-3">Case: {{case.display_name}}</h4><p class="text-muted {% if case
...
+{% macro roh_panel() %}
+<div class="panel panel-default" id="rohPanel">
+  <div data-toggle='tooltip' class="panel-heading" title="down format" >
+    Ideogram, UPD, ROH
+  </div>
+  <div class="panel-body">
+    <svg id="roh_svg" width="1500" height="1250"></svg> 
+  </div>
+  <div class="panel-footer">
+    <button class="btn btn-default form-control" data-toggle="modal" data-target="#edit-synopsis">Edit</button>
```
[the remark this answers is not in the document]
Comment (mikaell) [reply scout/server/blueprints/cases/templates/cases/case.html]: Fixed

**Comment by mikaell** [comment]
> * [x]  the dynamic sidebar doesn't play too well with this panel - it is all too happy to stay on top instead of forcing its space and the panel to shrink/rearrange into fewer columns.

I've pushed a fix. But I need help testing it -it looks ok now on my 13" screen- but I can't rescale to three columns. If any problems persists maybe it is possible to play around with the limits in the function `window.onresize=function(){...}`

**Comment by northwestwitch** [comment]
How is it going with this PR, the current changelog is so thick that I think it's about time to release a new version of Scout (what about early next week?). It would be nice to have this feature in it!

**Comment by dnil** [comment]
😄 If I recall correctly we decided to have this finished on 15 nov, after finishing the "Fix small things" milestone last Friday.

**Comment by northwestwitch** [comment]
> 😄 If I recall correctly we decided to have this finished on 15 nov, after finishing the "Fix small things" milestone last Friday.

Right. I'm just wondering if we should just merge when this is ready. Because it will be more and more complicated to keep it up to date as we go.

**Comment by dnil** [comment]
Ok, this is working, and does not interfere with operations when no chromograph files are loaded.
There are still issues with priority between the sidebar and svg image for certain size ranges, since both use absolute pixels and the responsive layout isn't quite configured to play well with this. I think one could find container settings to cope, but am ready to say we could merge and just make that an issue. E.g. when anyway adding coverage wigs as IGV tracks which is foreseen in the project.

**Comment by northwestwitch** [comment]
Code from this pull request shouldn't affect any case which doesn't have wig tracks right? If it is so I also agree to merge

**Comment by dnil** [comment]
That is indeed the intention. I fixed one more such case just now, and current tests pass.

**Comment by dnil** [comment]
Let’s go and adjust as needed later. It will certainly be easier to manage.

**Pull request merged**